### PR TITLE
refactor(k8s): refactor k8s code to java flavor

### PR DESCRIPTION
**Pull request by anda-ren**
## Description
- refactor k8s code to java flavor
## Modules
- [ ] UI
- [x] Controller
- [ ] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sClient.java [hunk 1]
```diff
@@ -23,26 +23,37 @@
...
      * Basic constructor for Kubernetes
      */
-    public Kubernetes(String ns) throws IOException {
-        client = Config.defaultClient();
+    public K8sClient(@Value("${sw.infra.k8s.name-space}") String ns) throws IOException {
+        String kubeConfigPath = System.getenv("HOME") + "/.kube/config";
+
+        // loading the out-of-cluster config, a kubeconfig from file-system
+        client =
+            ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(new FileReader(kubeConfigPath))).build();
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sClient.java:56]: `Config.defaultClient()`  will do the right actions,  ref: https://github.com/kubernetes-client/java/blob/f8a0654f355878ee0325227c28ca43968933db1e/util/src/main/java/io/kubernetes/client/util/ClientBuilder.java#L103

* KUBECONFIG env
* $HOME/.kube/config
* auto mounted ca
* etc

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+            try {
+                k8sClient.deleteJob(id.toString());
+            } catch (ApiException e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    private void deployTaskToK8s(K8sClient client, String image, TaskTrigger task) {//TODO TaskTrigger to Task
+        log.warn("aha {} {} {}", task.getId(), task.getResultPath(), task.getTaskType());
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java:83]: Refine the legacy debug log

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+            if (task.getTaskType() == TaskType.CMP) {
+                cmd = "cmp";
+            }
+            V1Job job = client.renderJob(getJobTemplate(), task.getId().toString(), "worker", image, List.of(cmd), envs);
+            // set result upload path
+            job.getSpec().getTemplate().getSpec().getContainers().get(0).env(List.of(new V1EnvVar().name("DST").value(prefix+ task.getResultPath().resultDir())
+                ,new V1EnvVar().name("MINIO_SERVICE").value(storageProperties.getS3Config().getEndpoint())
+                ,new V1EnvVar().name("MINIO_ACCESS_KEY").value(storageProperties.getS3Config().getAccessKey())
+                ,new V1EnvVar().name("MINIO_SECRET_KEY").value(storageProperties.getS3Config().getAccessKey())
+                )
```
Comment (jialeicui) [inline server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java:106]: `secretKey` ?

**Comment by jialeicui** [comment]
LGTM

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java (hunk 2 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java]: done

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java (hunk 3 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sTaskScheduler.java:106]: fixed

File: server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sClient.java (hunk 1 above)
Comment (anda-ren) [reply server/controller/src/main/java/ai/starwhale/mlops/schedule/k8s/K8sClient.java]: done

**Pull request merged**